h/browser/chrome/lib/uri-info.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * A wrapper for the server's uriinfo endpoint.
+ *
+ * Exports a function that makes an HTTP request to the server's uriinfo
+ * endpoint and returns a Promise that resolves to the response. Usage:
+ *
+ *     var uriInfo = require('uri-info');
+ *     uriInfo(uri).then(function(info) {
+ *       ...
+ *     });
+ *
+ * This function can be called with the same uri multiple times consecutively
+ * and it'll just keep returning the same Promise - won't make multiple HTTP
+ * requests.
+ *
+ * But if you call the function with a different uri, then call it with the
```

Comment:
Ok, there seems to be a problem with a race condition between tabs:

- Open tab 1 to uri 1, uriinfo request is sent
- Open tab 2 to uri 2, uriinfo request is sent
- Tab 2's uriinfo response is received
- Tab 1's uriinfo response is received

Both tabs will show the correct badge number, but if you click to activate tab 1 it will unnecessarily send the uriinfo request again. That's the only thing that goes wrong as far as I can tell. It does seem like the Promise should be per tab instead of global...